src/pages/index.js

```diff
@@ -1,96 +1,231 @@
+      {
+        data: cardData,
+        renderer: renderCard,
+      },
+      cardListSelector
+    );
+    sectionInstance.renderItems();
+  })
+  .catch(console.error);
+
+const modalWithImage = new PopupWithImage({
+  popupSelector: imageModalSelector,
+});
+
+const changeProfilePopup = new PopupWithForm({
+  popupSelector: modalChangeProfileSelector,
+  handleFormSubmit: (data) => {
+    changeProfilePopup.renderLoading(true);
+    api
+      .updateUserProfile({ avatar: data.url })
+      .then((data) => {
+        userInfo.setAvatartInfo(data.avatar);
+        changeProfilePopup.close();
+      })
+      .catch(console.error)
+      .finally(() => {
+        changeProfilePopup.renderLoading(false);
+      });
+  },
+  loadingText: "Saving...",
+});
+
+const modalWithFormUser = new PopupWithForm({
+  popupSelector: profileModalSelector,
+  handleFormSubmit: (data) => {
+    modalWithFormUser.renderLoading(true);
+    api
+      .userEditProfile(data)
+      .then((data) => {
+        userInfo.setUserInfo({
+          title: data.name,
+          description: data.about,
+        });
+        userInfo.setAvatartInfo(data.avatar);
+        modalWithFormUser.close();
+      })
+      .catch(console.error)
+      .finally(() => {
+        modalWithFormUser.renderLoading(false);
+      });
+  },
+  loadingText: "Saving...",
+});
+
+const confirmModal = new PopupWithForm({
+  popupSelector: cardDeleteModalSelector,
+  loadingText: "Deleting...",
+});
+
+const modalWithFormImage = new PopupWithForm({
+  popupSelector: cardModalSelector,
+  handleFormSubmit: (data) => {
+    modalWithFormImage.renderLoading(true);
+    api
+      .addCard(data)
+      .then((data) => {
+        renderCard(data);
+        modalWithFormImage.close();
+      })
+      .catch(console.error)
+      .finally(() => {
+        modalWithFormImage.renderLoading(false);
+      });
+  },
+  loadingText: "Saving...",
+});
 
 function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleCardClick);
-  const cardElement = card.createCardElement();
-  cardSection.addItem(cardElement);
+  const cardImage = createCard(cardData);
+  sectionInstance.prependItem(cardImage);
 }
 
-function handleCardClick(data) {
-  popupImage.open(data);
+// setting event listeners
+modalWithFormUser.setEventListeners();
+modalWithFormImage.setEventListeners();
+modalWithImage.setEventListeners();
+confirmModal.setEventListeners();
+changeProfilePopup.setEventListeners();
+
+function createCard(cardData) {
+  const card = new Card(
+    {
+      cardData,
+      myId: userId,
+      handleImageClick: (data) => {
+        modalWithImage.open(data);
+      },
+      handleDeleteClick: () => {
+        confirmModal.open();
+        confirmModal.setSubmitAction(() => {
+          confirmModal.renderLoading(true);
+          const id = card.getId();
+          api
+            .removeCard(id)
+            .then(() => {
+              card.handleDeleteIcon();
```

Comment:
This method should be called `deleteCard` or something like that, since it's not an event listener anymore